File: client/src/api/characters/CharacterDetailResponse.ts
import type { RuleSnapshotDto } from '../rules/RuleSnapshotDto'
import type { CalculatedCharacterStatsDto } from './CalculatedCharacterStatsDto'
import type { CharacterDraftProfileResponse } from './CharacterDraftProfileResponse'
import type { CharacterSkillResponse } from './CharacterSkillResponse'
import type { CharacterSummaryResponse } from './CharacterSummaryResponse'

export type CharacterDetailResponse = CharacterSummaryResponse & {
  ruleSnapshot: RuleSnapshotDto | null
  calculatedStats: CalculatedCharacterStatsDto | null
  draftProfile: CharacterDraftProfileResponse | null
  skills: CharacterSkillResponse[]
}

File: server/src/GenesysForge.Contracts/Characters/CharacterDetailResponse.cs
using GenesysForge.Contracts.Rules;

namespace GenesysForge.Contracts.Characters;

public sealed record CharacterDetailResponse(
    Guid Id,
    string Name,
    CharacterStatus Status,
    Guid RulesetId,
    DateTimeOffset UpdatedAt,
    RuleSnapshotDto? RuleSnapshot,
    CalculatedCharacterStatsDto? CalculatedStats);
